examples/kitchen-sink/server/controllers/home.js

```diff
@@ -64,6 +67,7 @@ module.exports = (_, response, next) => {
             </section>
           </div>
         </Layout>
+        <DataEmbed id={DATA_EMBED_ID} data={{ foo: true, bar: 'qux' }} />
```

Comment:
There can be multiple instances of `DataEmbed` if a user wants, so right now it's up to them how to get data to DataEmbed.

In theory we could replicate the middleware and server packages of app-context but that felt more than what was initially agreed in the original issue.

Then DataEmbed would just be a generic AppContext.

Reply:
> In theory we could replicate the middleware and server packages 

Feels like over-engineering? 😬 We don't expect many apps to interact with this tool directly

Reply:
I've pushed a commit to manually move the data into the controller: https://github.com/Financial-Times/dotcom-page-kit/pull/812/commits/3204cf6d5748a985b1f977d290cfdc943545c0de

Reply:
> There can be multiple instances of DataEmbed

So using [this usage of Page Kit](https://github.com/Financial-Times/next-subscribe/blob/85fc96193ae6446d3f76f27e6a04248445e9a0f3/server/page-kit-wrapper.js#L50-L54) as an example, it would look something like:

```
const document = React.createElement(
	Shell,
	{ ...shellProps },
	[
		React.createElement(Layout, { ...layoutProps, contents: html }),
		React.createElement(ClientEmbed, id='custom-context-foobar' data={{ foo: true, bar: 'qux' }}),
		React.createElement(ClientEmbed, id='custom-context-barfoo' data={{ foo: false, bar: 'quux' }})
	]
);
```

- This will require the document element created by the consuming app to have that many more arguments (at the moment it is quite clean with just `Shell` and `Layout`), but admittedly it is currently only one app that will be using `ClientEmbed` so is not a huge consideration.
- In keeping to the principle of least surprise, users may well expect for all the client-embedded data (flags, appContext, custom) to sit alongside one another, and that this could be achieved by passing the custom data as part of the props given to the `Shell` component - is this easily possible?

Reply:
>  is this easily possible?

I think it is, but we'd want to mirror the `dotcom-server-app-context` and `dotcom-middleware-app-context` package functionality so that it is integrated in a similar way to app context which we discussed above.

So we could go ahead and flesh that functionality out, or intentionally keep this simpler and observe how people use the package before committing to something in page kit.

Reply:
👍 Okay - let's keep it as is and monitor.